File: Safari-609.1.11.1/LayoutTests/http/tests/media/resources/video-auth.php
<?php

    $ua = $_SERVER["HTTP_USER_AGENT"];

	header("Cache-Control: no-store");
	header("Connection: close");
	if (!isset($_SERVER['PHP_AUTH_USER'])) {
	    header("WWW-authenticate: Basic realm=\"" . $_SERVER['REQUEST_URI'] . "\"");
	    header('HTTP/1.0 401 Unauthorized');
	    exit;
	}

    $fileName = $_GET["name"];
    $type = $_GET["type"];

    $_GET = array();
    $_GET['name'] = $fileName;
    $_GET['type'] = $type;
    @include("./serve-video.php"); 
?>

File: LayoutTests/http/tests/media/resources/video-auth.py
#!/usr/bin/env python3

import base64
import os
import sys
from urllib.parse import parse_qs, urlencode

username = base64.b64decode(os.environ.get('HTTP_AUTHORIZATION', ' Og==').split(' ')[1]).decode().split(':')[0]
query = parse_qs(os.environ.get('QUERY_STRING', ''), keep_blank_values=True)

sys.stdout.write(
    'Cache-Control: no-store\r\n'
    'Connection: close\r\n'
)

if not username:
    sys.stdout.write(
        'WWW-authenticate: Basic realm="{}"\r\n'
        'status: 401\r\n\r\n'.format(os.environ.get('REQUEST_URI', ''))
    )
    sys.exit(0)

filename = query.get('name', ['404.txt'])[0]
typ = query.get('type', [''])[0]

os.environ['QUERY_STRING'] = urlencode({'name': filename, 'type': typ})

from serve_video import serve_video
